### PR TITLE
Fixes 4762: Add missing nil checks for pulp methods

### PR DESCRIPTION
**Pull request by rverdile**
## Summary
I could not reproduce the original issue, but this adds checks to places I believe could cause a panic under the right conditions.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4762

File: pkg/pulp_client/uploads.go [hunk 1]
```diff
@@ -28,30 +28,34 @@ func (r *pulpDaoImpl) CreateUpload(ctx context.Context, size int64) (*zest.Uploa
...
-		return &zest.UploadResponse{}, httpResp.StatusCode, errorWithResponseBody("uploading file chunk", httpResp, err)
+		return &zest.UploadResponse{}, statusCode, errorWithResponseBody("uploading file chunk", httpResp, err)
 	}
-	return readResp, httpResp.StatusCode, nil
+	return readResp, statusCode, nil
 }
 
 // FinishUpload Finishes an upload
 func (r *pulpDaoImpl) FinishUpload(ctx context.Context, uploadHref string, sha256 string) (*zest.AsyncOperationResponse, int, error) {
+	var statusCode int
```
Comment (dominikvagner) [inline pkg/pulp_client/uploads.go:47]: same as the comment above 🤔😅

**Review by dominikvagner — COMMENTED** [review]
one small question, otherwise looks good! 👏🏼

File: pkg/pulp_client/uploads.go [hunk 2]
```diff
@@ -28,30 +28,34 @@ func (r *pulpDaoImpl) CreateUpload(ctx context.Context, size int64) (*zest.Uploa
 
 // UploadChunk Uploads a chunk for an upload
 func (r *pulpDaoImpl) UploadChunk(ctx context.Context, uploadHref string, contentRange string, file *os.File, sha256 string) (*zest.UploadResponse, int, error) {
-	ctx, client := getZestClient(ctx)
+	var statusCode int
```
[the remark this answers is not in the document]
Comment (rverdile) [reply pkg/pulp_client/uploads.go:31]: Good question. I would say the status code here is meant to be whatever is returned by the pulp API. In my opinion, if the pulp API does not return a status code, then this method shouldn't either (i.e. return 0)

[the remark this answers is not in the document]
Comment (rverdile) [reply pkg/pulp_client/uploads.go:31]: But we should make sure that our API does return the correct error code, so I'll double check that behavior

[the remark this answers is not in the document]
Comment (rverdile) [reply pkg/pulp_client/uploads.go]: I realized we pass that statusCode straight through our API, so I implemented your suggestion. Thanks!

**Review by dominikvagner — APPROVED** [review]
everything looks good now 🎉 
approved ✅ nice work!

**Pull request merged**